egui_extras/src/table.rs

```diff
@@ -352,24 +352,144 @@ pub struct TableBody<'a> {
+    /// ### Example
+    /// ```
+    /// # egui::__run_test_ui(|ui| {
+    /// use egui_extras::{TableBuilderSize};
+    /// TableBuilder::new(ui)
+    ///     .column(Size::remainder().at_least(100.0))
+    ///     .body(|mut body| {
+    ///         let row_heights: Vec<f32> = vec![60.0, 18.0, 31.0, 240.0];
+    ///         body.heterogeneous_rows(row_heights.iter(), |row_index, mut row| {
+    ///             let thick = row_index % 6 == 0;
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.label(row_index.to_string());
+    ///                 });
+    ///             });
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.label(clock_emoji(row_index));
+    ///                 });
+    ///             });
+    ///             row.col(|ui| {
+    ///                 ui.centered_and_justified(|ui| {
+    ///                     ui.style_mut().wrap = Some(false);
+    ///                     if thick {
+    ///                         ui.heading("Extra thick row");
+    ///                     } else {
+    ///                         ui.label("Normal row");
+    ///                     }
+    ///                 });
+    ///             });
+    ///         });
+    ///     });
+    /// # });
+    /// ```
+    pub fn heterogeneous_rows(
+        &mut self,
+        heights: impl Iterator<Item = f32>,
+        mut populate_row: impl FnMut(usize, TableRow<'_, '_>),
+    ) {
+        // VIRTUAL_EXTENSION represents how far above and below the limits of the visible rectangle
+        // that we should consider when determining which rows to actually populate. this provides
+        // the illusion of table rows sliding out of view (rather than disappearing abruptly)
+        const VIRTUAL_EXTENSION: f32 = 30.0;
+
+        // in order for each row to retain its striped color as the table is scrolled, we need an
+        // iterator with the boolean built in based on the enumerated index of the iterator element
+        let mut striped_heights = heights
+            .enumerate()
+            .map(|(index, height)| (index, index % 2 == 0, height));
+
+        let max_height = self.end_y - self.start_y + VIRTUAL_EXTENSION;
+        let y_progress = self.y_progress() - VIRTUAL_EXTENSION;
+
+        // cumulative height of all rows above those being displayed
+        let mut height_above_visible: f64 = 0.0;
+        // cumulative height of all rows below those being displayed
+        let mut height_below_visible: f64 = 0.0;
+
+        // calculate height above visible table range
+        while let Some((row_index, striped, height)) = striped_heights.next() {
+            // when y_progress is greater than height above 0, we need to increment the row index
+            // and update the height above visble with the current height then continue
+            if height_above_visible >= y_progress as f64 {
```

Comment:
I believe `VIRTUAL_EXTENSION` could be removed if we changed this logic to take `height` into account (`if height_above_visible + height as f64 >= …`).

(PS: I thought I already made this comment, but now I can't find it on GitHub…)